vespaclient-java/src/main/java/com/yahoo/vespavisit/VdsVisitHandler.java

```diff
@@ -30,14 +34,26 @@ public abstract class VdsVisitHandler {
     String lastPercentage;
     final Object printLock = new Object();
 
-    protected String progressFileName = "";
+    private static class ProgressMeta {
+        String fileName = "";
+        String lastProgressContents;
+        int unwrittenUpdates = 0;
+        long lastWriteAtNanos = 0;
+        Duration writeInterval = Duration.ofSeconds(10);
```

Comment:
Progress is automatically flushed on `SIGINT` and other non-fatal visitor terminations, so I think <=10 seconds worth of potential duplicate results on a hard crash is an okay tradeoff since there will be duplicate results from every bucket in flight anyway.